### ISSUE TITLE
Re-requesting an Existing Pull Request

### ISSUE THREAD

**Issue by sky-joker** [ansible-collections/community.zabbix#34]
##### SUMMARY

The following are existing pull requests for the Zabbix module in ansible upstream that I have now confirmed.

- [x] ~~https://github.com/ansible/ansible/pull/68333~~
- [x] ~~https://github.com/ansible/ansible/pull/66860~~
- [x] ~~https://github.com/ansible/ansible/pull/64607~~
- [x] ~~https://github.com/ansible/ansible/pull/64428~~
- [x] https://github.com/ansible/ansible/pull/31933
- [x] ~~https://github.com/ansible/ansible/pull/58051~~
- [x] https://github.com/ansible/ansible/pull/43311

If @D3DeFi agrees to https://github.com/ansible-collections/community.zabbix/issues/33#issuecomment-623140548, let's re-ask for the above Pull Request.  
If you agree, would you like to request all the above PR again?

**Comment by D3DeFi** [comment]
@sky-joker I agree. If you want, I can iterate through them tomorrow, request each to be reopened here and close them in ansible/ansible.

**Comment by sky-joker** [comment]
Thank you @D3DeFi 

> I agree. If you want, I can iterate through them tomorrow, request each to be reopened here and close them in ansible/ansible.

Could you please handle this for me?

**Comment by D3DeFi** [comment]
I believe that everything I've had tagged as a todo is now closed in ansible/ansible.

Added a few PRs to your list as well.

**Comment by sky-joker** [comment]
Thank you @D3DeFi for your cooperation!
I checked the box for closed ones.

**Comment by D3DeFi** [comment]
Np, but I would still leave this issue open until we are sure that each of those PRs has been reopened here.

**Comment by D3DeFi** [comment]
ansible/ansible#64428 merged to the collection as #68

**Comment by D3DeFi** [comment]
both ansible/ansible#64607 and ansible/ansible#64428 deprecated by ansible/ansible#67693 and #82

**Comment by D3DeFi** [comment]
@sky-joker I was looking at ansible/ansible#66860. In the first commit, the patch started as something different and actually introduced something, but after 11 commits later (that the patch consists of) the actual changes (logical) are quite minimal. 

Basically this:
```python
if inventory_mode:
    if LooseVersion(self._zbx_api_version) <= LooseVersion('4.4.0'):
        if host['inventory']:
            if int(host['inventory']['inventory_mode']) != self.inventory_mode_numeric(inventory_mode):
                return True
        elif inventory_mode != 'disabled':
            return True
    else:
        if int(host['inventory_mode']) != self.inventory_mode_numeric(inventory_mode):
            return True
```

Versus this (exact same thing with additional comment, changed positions of if statements and version bump from 4.4.0 to 4.4.1):
```python
if inventory_mode:
    if host['inventory']:
        # The way to read the inventory_mode is different on versions older than 4.4.1
        if LooseVersion(self._zbx_api_version) < LooseVersion('4.4.1'):
            if int(host['inventory']['inventory_mode']) != self.inventory_mode_numeric(inventory_mode):
                return True
        else:
            if int(host['inventory_mode']) != self.inventory_mode_numeric(inventory_mode):
                return True
    elif inventory_mode != 'disabled':
        return True
```

I don't see any meaning in reordering of `if` statements, neither do we miss on comment. As for the version bump `4.4.0` -> `4.4.1` should I commit that? Is someone even using 4.4.1 anymore?

**Comment by sky-joker** [comment]
Thank you @D3DeFi for looking at https://github.com/ansible/ansible/pull/66860.
hm…This is a tough one... 

The Zabbix 4.4 will no longer be supported after July.
https://www.zabbix.com/life_cycle_and_release_policy

Because of that, I wonder we don’t need the support of a `zabbix_host` module.
However, I think Zabbix 4.4.1 should also be considered if `zabbix_host` will be supporting Zabbix 4.4.
Because the return JSON structure has changed from 4.4.1.
https://github.com/ansible/ansible/pull/66860#discussion_r387741567

**Comment by D3DeFi** [comment]
I wanted to commit that, but I've just spotted another difference in the snippets above (comparison character):
`    if LooseVersion(self._zbx_api_version) <= LooseVersion('4.4.0'):`
vs:
`    if LooseVersion(self._zbx_api_version) < LooseVersion('4.4.1'):`

So I think that we already support 4.4.1 properly. Judging from the source code you have linked in the previous conversation [4.4.0](https://github.com/zabbix/zabbix/blob/4.4.0/frontends/php/host_prototypes.php#L287) vs [4.4.1](https://github.com/zabbix/zabbix/blob/4.4.1/frontends/php/host_prototypes.php#L284) this further confirms that. Thanks! :)

Next on my radar would be ansible/ansible#58051. Do you have something planned for it or can I work towards porting it here?

**Comment by sky-joker** [comment]
> Next on my radar would be ansible/ansible#58051. Do you have something planned for it or can I work towards porting it here?

I don’t have a plan, so would you create this patch and PR if possible?
If you busy, I’ll start to work on this patch creation :)

**Comment by D3DeFi** [comment]
> 
> 
> > Next on my radar would be [ansible/ansible#58051](https://github.com/ansible/ansible/pull/58051). Do you have something planned for it or can I work towards porting it here?
> 
> I don’t have a plan, so would you create this patch and PR if possible?
> If you busy, I’ll start to work on this patch creation :)

No worries. I will try some cherry-pick magic to see if we can preserve original author contributions.

**Comment by JiffsMaverick** [comment]
@D3DeFi @sky-joker Hello! Is it possible to backport support for version 5.0 of Zabbix to ansible 2.9? You fixed these bugs in 2.10 and it was awesome, thanks. But we have been unable to work normally with Zabbix 5.0 on ansible 2.9 for several months due to this problem.

PS: it seems that for some reason, this bugfix (ansible/ansible@7b2cfdacd00ddf907247270d228a6bf5f72258a1 and ansible/ansible#65392) did not get into the 2.9 branch.

**Comment by D3DeFi** [comment]
@JiffsMaverick boi, 9 thumbsups after 6 hours? Why noone requested this sooner :( With sooo much happening around collections these days (well since march) we've completely forgot about backporting to old ansible. I am sorry.

For 5.0 zero you would require even #51 and that may be considered a feature request, I don't know if it would be accepted, but the one above mentioned should get through. I will try to find some time during weekend hopefully.

Btw putting this [version of zabbix_host](https://github.com/ansible-collections/community.zabbix/blob/82e5df6930f26e59e6ae82e6b51aca06444edd2c/plugins/modules/zabbix_host.py) into `library/` folder should work as a workaround. We can dig for other modules in the commit as well if you need those too.

**Comment by D3DeFi** [comment]
@JiffsMaverick do you please have a possibility to test [ansible/ansible#71289](https://github.com/ansible/ansible/pull/71289) within your environment and comment on the (back)port PR that it is working for you?

**Comment by JiffsMaverick** [comment]
@D3DeFi Thank you for fast fix! Changes from #71288 and #71289 works!

**Comment by D3DeFi** [comment]
Seems that  ansible/ansible#71289 won't be merged, but with ansible/ansible#71288 it seems to no longer fail on Zabbix 5.0 (even though you will not be able to manage details for SNMP interfaces. So next minor release of 2.9.x should have this fixed.

**Comment by derekpurdy** [comment]
> Btw putting this [version of zabbix_host](https://github.com/ansible-collections/community.zabbix/blob/82e5df6930f26e59e6ae82e6b51aca06444edd2c/plugins/modules/zabbix_host.py) into `library/` folder should work as a workaround. We can dig for other modules in the commit as well if you need those too.

I am having the same issue and I'm just confused on how to use this workaround. Where do you place that file?

**Comment by D3DeFi** [comment]
@derekpurdy if you have ansible 2.9 failing with KeyError on zabbix 5.0 then you should be fine with just updating to ansible 2.9.13 as some minor bugfixes were included in the latest [release](https://github.com/ansible/ansible/blob/stable-2.9/changelogs/CHANGELOG-v2.9.rst).


> > Btw putting this [version of zabbix_host](https://github.com/ansible-collections/community.zabbix/blob/82e5df6930f26e59e6ae82e6b51aca06444edd2c/plugins/modules/zabbix_host.py) into `library/` folder should work as a workaround. We can dig for other modules in the commit as well if you need those too.
> 
> 
> 
> I am having the same issue and I'm just confused on how to use this workaround. Where do you place that file?


This is somewhat ugly hack used for official modules if you need to override them. More information can be found in [this documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_locally.html).

To sum it up. You just create _library_ folder inside your project directory and download ansible module to it. Then if you run ansible with verbose argument you should be able to see that it used local module instead of the one shipped with it.

You just need to remember the reason why you did it an remove locally stored module once the fiz is available in some release to prevent the module not being updated in the future.

Btw, since ansible 2.9.10 you should by able to also [install collections](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html) as a whole. This is probably the best course of action as ansible 2.10 will ship collections by default and relevant fixes and features will be backported to 2.9 less and less.

**Comment by D3DeFi** [comment]
https://github.com/ansible/ansible/pull/43311 - cannot apply any cherry-pick magic and original author not responding so this will not be moved to the collection.
https://github.com/ansible/ansible/pull/31933 - same
